### ISSUE TITLE
Hanging on "[CodePush] Checking for update."

### ISSUE THREAD

**Issue by morelazers** [Microsoft/react-native-code-push#513]
`MainApplication.java`:

```
new CodePush("MY_STAGING_DEPLOYMENT_KEY", getApplicationContext(), BuildConfig.DEBUG)
```
My root component:

```
const codepushOptions = { updateDialog: true, installMode: codePush.InstallMode.IMMEDIATE };
var MyApp = codePush(codepushOptions)(Wrapper);
export default MyApp;
```

I've released both staging & production versions, but when I look at the console output when I am running the app, all I see is `[CodePush] Checking for update.`

If I inspect the request that the app sends, I get this:
```
updateInfo": {
		"downloadURL": "A_VALID_LINK",
		"description": "",
		"isAvailable": true,
		"isMandatory": true,
		"appVersion": "1.0.0",
		"packageHash": "A_HASH",
		"label": "v34",
		"packageSize": 351745,
		"updateAppVersion": false,
		"shouldRunBinaryVersion": false
	}
```

If I try to call `codePush.sync()` manually anywhere anywhere in the app after this, I get a message that states that sync is already underway.

Am I doing something wrong? Any help is super appreciated :))

**Comment by morelazers** [comment]
I have narrowed this down to something within my app. I have overridden `window.XMLHttpRequest` to solve another issue, and it is this which is causing the issue.

I will close this issue, since it is not a problem with this package.

**Comment by geof90** [comment]
Thanks! Glad you are unblocked.
